File: _1.20/src/main/java/cn/enaium/noexpensive/command/ReloadCommand.java
package cn.enaium.noexpensive.command;

import cn.enaium.noexpensive.Config;
import com.mojang.brigadier.Command;
import com.mojang.brigadier.CommandDispatcher;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.text.Text;

import static cn.enaium.noexpensive.NoExpensive.ROOT;

/**
 * @author Enaium
 */
public class ReloadCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(ROOT.then(CommandManager.literal("reload").executes(context -> {
            Config.load();
            context.getSource().sendFeedback(() -> Text.translatable("command.reload.success"), false);
            return Command.SINGLE_SUCCESS;
        })));
    }
}

File: modern/_1.16.5/src/main/kotlin/cn/enaium/noexpensive/command/ReloadCommand.kt
package cn.enaium.noexpensive.command

import cn.enaium.noexpensive.Config
import cn.enaium.noexpensive.ROOT
import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.TranslatableText

/**
 * @author Enaium
 */
fun reloadCommand(dispatcher: CommandDispatcher<ServerCommandSource>) {
    dispatcher.register(
        ROOT.then(
            CommandManager.literal("reload").executes { context: CommandContext<ServerCommandSource> ->
                Config.load()
                context.source.sendFeedback(TranslatableText("command.reload.success"), false)
                Command.SINGLE_SUCCESS
            })
    )
}
